Sources/SPTDataLoaderURLCache.m

```diff
@@ -0,0 +1,60 @@
+/*
+ Copyright (c) 2015-2019 Spotify AB.
+
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+ */
+#import "SPTDataLoaderURLCache.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SPTDataLoaderURLCache ()
+
+@end
+
+@implementation SPTDataLoaderURLCache
+
+#pragma mark SPTDataLoaderURLCache
+
+- (NSCachedURLResponse *)addCachedHeaderToResponse:(NSCachedURLResponse *)cachedResponse
+{
+    if ([cachedResponse.response isKindOfClass:[NSHTTPURLResponse class]]) {
+        NSHTTPURLResponse *response = (NSHTTPURLResponse*) cachedResponse.response;
+        NSMutableDictionary *headers = [response.allHeaderFields mutableCopy];
+        headers[@"Cached"] = @"YES";
```

Comment:
For the header name I would use a more weird string to avoid clashing with an existent one. Something like @"X-SPTDataLoader-Cache-Header" and put it as a constant in a shared header file.

Reply:
updated.